### DISCUSSION TITLE
Multitenancy?

### DISCUSSION THREAD

**Discussion by dasfliege** [nautobot/nautobot#5252] [General]
Just stumbled across nautobot while searchin for an alternative to netbox that is suitable for service providers.
We have a quite large customer base and are searching for a tool that allows us to manage all DC's that we are responsible for. That includes physical infrastructure as well as IPAM and network documentation. Netbox is quite what we are searching for, except that it does not fullfill our needs in terms of multitenancy. Even though we assign all object to a tenant, building permissions models with constraints is quite a pain and does not cover everything. F.e. VLAN groups are not assignable to a tenant. And there are quite a few other things that shows that netbox has not been built with multitenancy support in mind. Tenant ID are names differently in every module. You have to create several constraints because of that to do proper filtering. And even then, it does not cover everything.

Does nautobot has a proper implementation for multitenancy that includes each an everything?

**Comment by lampwins** [comment]
Hi, today Nautobot’s tenant implementation closely follows that of NetBox, but we do have plans to expand tenant support to all parts of the data model, and to build simplifications into the permission system so that constraints can more easily be built for these types of MSP environments with tenants in heavy use. This is something we are looking at including in the v3 timeframe, later this year.

**Reply by dasfliege** [reply]
Thanks a lot for your answer. That sounds promising, as netbox does not seem to have any interest in expanding the datamodel to better suit MSP needs. I was already thinking about developing another tailored solution for our purposes, but may try nautobot instead now.

**Reply by dasfliege** [reply]
Guys... Installed nautobot yesterday and tried a few things today. I have to say that this is just so much better than netbox. Everything makes so much more sense. Pretty much all the things i struggled with in netbox, like missing logics for active/passive cluster, missing namespaces, etc. is just here, in quite a similar way that i would have implemented it. It's pretty obvious to me, that the guys forked this, have a much better understanding of how a proper network design should look like. Thanks a lot for this! Looking forward to have proper multi tenancy support though ;-)

**Comment by dasfliege** [comment]
I wanted to followup on this. We're heavily using Nautobot for our own datacenter are extremely happy with it. But we're waiting for the multitenancy implementation to also use it for all our on-prem customers. Are there any news regarding the implementation? Still considered for v3? And is there any estimation for the release of v3?

**Comment by dasfliege** [comment]
No one? :-(

**Comment by dasfliege** [comment]
So v3 has been released but still no proper multitenancy support implemented as far as i can read in the releasenotes :-( Has it been rescheduled to v4?

**Comment by diogenes00** [comment]
That's correct, multitenancy was *not* part of the 3.0 release. As mentioned above, there were talks about adding that functionality -- as well as improving the permissions user experience -- but they did not actually make the cut for 3.0, unfortunately. As of the near future (2026 plan), we don't yet have it as part of a plan but would be happy to discuss specifics.

**Reply by dasfliege** [reply]
Thanks for the answer.
Just let me know if we can assist in outlining the requirements and functions for a proper multitenancy implementation.
